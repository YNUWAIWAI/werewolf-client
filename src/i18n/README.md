# i18n

## Message Syntax
https://formatjs.io/guides/message-syntax/

## id

`()`内の値は実行時に決定されることを示す．
