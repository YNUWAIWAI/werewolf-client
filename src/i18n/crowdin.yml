project_identifier_env: CROWDIN_PROJECT_ID
api_key_env: CROWDIN_API_KEY
base_path : "./"
preserve_hierarchy: true

files:
  -
    source: '*/en.json'
    translation: '%original_path%/%locale%.json'
    languages_mapping:
      locale:
        'af': 'af'
        'ar': 'ar'
        'bs-BA': 'bs-BA'
        'ca': 'ca'
        'cs': 'cs'
        'da': 'da'
        'de': 'de'
        'el': 'el'
        'es-ES': 'es-ES'
        'fa': 'fa-IR'
        'fi': 'fi'
        'fr': 'fr'
        'he': 'he'
        'hu': 'hu'
        'id': 'id-ID'
        'it': 'it'
        'ja': 'ja'
        'ko': 'ko'
        'mr': 'mr-IN'
        'nl': 'nl'
        'no': 'no-NO'
        'pl': 'pl'
        'pt-BR': 'pt-BR'
        'pt-PT': 'pt-PT'
        'ro': 'ro'
        'ru': 'ru'
        'sk': 'sk-SK'
        'sr': 'sr'
        'sv-SE': 'sv-SE'
        'tr': 'tr'
        'uk': 'uk'
        'vi': 'vi'
        'zh-CN': 'zh-CN'
        'zh-TW': 'zh-TW'
