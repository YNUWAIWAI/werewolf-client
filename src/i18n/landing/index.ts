import enMessages from './en'
import frMessages from './fr'
import itMessages from './it'
import jaMessages from './ja'

export const enum Language {
  en = 'en',
  fr = 'fr',
  it = 'it',
  ja = 'ja'
}
interface Messages {
  [id: string]: string
}

export const getMessages = (locale: Language): Messages => {
  switch (locale) {
    case 'en':
      return enMessages
    case 'fr':
      return frMessages
    case 'it':
      return itMessages
    case 'ja':
      return jaMessages
    default:
      throw Error(`${locale} is unsuported`)
  }
}
