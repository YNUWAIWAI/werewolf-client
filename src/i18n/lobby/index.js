// @flow
import type {Language} from 'village'
import en from './en'
import fr from './fr'
import it from './it'
import ja from './ja'

export const getMessages = (locale: Language) => {
  switch (locale) {
    case 'en':
      return en
    case 'fr':
      return fr
    case 'it':
      return it
    case 'ja':
      return ja
    default:
      throw Error(`${locale} is unsuported`)
  }
}
