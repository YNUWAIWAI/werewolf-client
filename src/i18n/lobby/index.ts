/// <reference types="lobby" />
import en from './en'
import fr from './fr'
import it from './it'
import ja from './ja'

export const getMessages = (locale: lobby.Language) => {
  switch (locale) {
    case 'en':
      return en
    case 'fr':
      return fr
    case 'it':
      return it
    case 'ja':
      return ja
    default:
      throw Error(`${locale} is unsuported`)
  }
}
