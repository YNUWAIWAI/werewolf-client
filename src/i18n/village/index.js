// @flow
import type {Language} from 'village'
import en from './en'
import ja from './ja'

export const getMessages = (locale: Language) => {
  switch (locale) {
    case 'en':
      return en
    case 'ja':
      return ja
    default:
      throw Error(`${locale} is unsuported`)
  }
}
