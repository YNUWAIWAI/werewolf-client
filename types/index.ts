import * as lobby from './lobby'
import * as village from './village'

export {
  lobby,
  village
}
