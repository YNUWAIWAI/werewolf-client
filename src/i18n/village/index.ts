import * as village from '../../../types/village'
import en from './en.json'
import fr from './fr.json'
import it from './it.json'
import ja from './ja.json'

export const getMessages = (locale: village.Language) => {
  switch (locale) {
    case 'en':
      return en
    case 'fr':
      return fr
    case 'it':
      return it
    case 'ja':
      return ja
    default:
      throw Error(`${locale} is unsuported`)
  }
}
