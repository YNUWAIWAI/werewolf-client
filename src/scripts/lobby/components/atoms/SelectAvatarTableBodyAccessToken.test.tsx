import * as React from 'react'
import Copy from './svg/Copy'
import SelectAvatarTableBodyAccessToken from './SelectAvatarTableBodyAccessToken'
import {getMessages} from '../../i18n'
import {initRenderer} from '../../tools'
import {lobby} from '../../types'

const {mountWithIntl} = initRenderer(lobby.Language.en, getMessages(lobby.Language.en))

describe('render', () => {
  test('navigator.clipboard !== undefined', () => {
    const navigatorClipBoard = navigator.clipboard;

    (navigator as any).clipboard = {}
    const handleSelect = jest.fn()
    const renewAccessToken = jest.fn()
    const wrapper = mountWithIntl(
      <SelectAvatarTableBodyAccessToken
        additionalClassName={[]}
        handleSelect={handleSelect}
        renewAccessToken={renewAccessToken}
        token="token"
      />
    )

    expect(wrapper.find(Copy).exists()).toBe(true);
    (navigator as any).clipboard = navigatorClipBoard
  })
  test('navigator.clipboard === undefined', () => {
    const handleSelect = jest.fn()
    const renewAccessToken = jest.fn()
    const wrapper = mountWithIntl(
      <SelectAvatarTableBodyAccessToken
        additionalClassName={[]}
        handleSelect={handleSelect}
        renewAccessToken={renewAccessToken}
        token="token"
      />
    )

    expect(wrapper.find(Copy).exists()).toBe(false)
  })
})
test('copy', () => {
  const navigatorClipBoard = navigator.clipboard
  const writeText = jest.fn();

  (navigator as any).clipboard = {
    writeText
  }
  const handleSelect = jest.fn()
  const renewAccessToken = jest.fn()
  const wrapper = mountWithIntl(
    <SelectAvatarTableBodyAccessToken
      additionalClassName={[]}
      handleSelect={handleSelect}
      renewAccessToken={renewAccessToken}
      token="token"
    />
  )

  wrapper.find('div.copy').simulate('click')
  expect(writeText).toHaveBeenCalledTimes(1)
  expect(writeText).toHaveBeenCalledWith('token');
  (navigator as any).clipboard = navigatorClipBoard
})
test('handleFocus', () => {
  const handleSelect = jest.fn()
  const renewAccessToken = jest.fn()
  const wrapper = mountWithIntl(
    <SelectAvatarTableBodyAccessToken
      additionalClassName={[]}
      handleSelect={handleSelect}
      renewAccessToken={renewAccessToken}
      token="token"
    />
  )
  const select = jest.fn()

  wrapper.find('input').simulate('focus', {
    target: {
      select
    }
  })
  expect(select).toHaveBeenCalledTimes(1)
})
describe('handleSelect', () => {
  test('event.target === event.currentTarget', () => {
    const handleSelect = jest.fn()
    const renewAccessToken = jest.fn()
    const wrapper = mountWithIntl(
      <SelectAvatarTableBodyAccessToken
        additionalClassName={[]}
        handleSelect={handleSelect}
        renewAccessToken={renewAccessToken}
        token="token"
      />
    )

    wrapper.find('.lo--select-avatar--table--body--item.access-token').simulate('click')
    expect(handleSelect).toHaveBeenCalledTimes(1)
  })
  test('event.target !== event.currentTarget', () => {
    const handleSelect = jest.fn()
    const renewAccessToken = jest.fn()
    const wrapper = mountWithIntl(
      <SelectAvatarTableBodyAccessToken
        additionalClassName={[]}
        handleSelect={handleSelect}
        renewAccessToken={renewAccessToken}
        token="token"
      />
    )

    wrapper.find('input').simulate('click')
    expect(handleSelect).toHaveBeenCalledTimes(0)
  })
})
test('renewAccessToken', () => {
  const handleSelect = jest.fn()
  const renewAccessToken = jest.fn()
  const wrapper = mountWithIntl(
    <SelectAvatarTableBodyAccessToken
      additionalClassName={[]}
      handleSelect={handleSelect}
      renewAccessToken={renewAccessToken}
      token="token"
    />
  )

  wrapper.find('.renew').simulate('click')
  expect(renewAccessToken).toHaveBeenCalledTimes(1)
})
