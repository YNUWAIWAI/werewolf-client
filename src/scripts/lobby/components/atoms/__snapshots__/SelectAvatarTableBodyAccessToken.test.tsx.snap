// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render navigator.clipboard !== undefined 1`] = `"<div class=\\"lo--select-avatar--table--body--item access-token \\"><input readonly=\\"\\" type=\\"text\\" value=\\"token\\"><div class=\\"copy\\"><svg class=\\" copy\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><path d=\\"M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z\\"></path></svg></div><div class=\\"renew\\">Renew</div></div>"`;

exports[`render navigator.clipboard === undefined 1`] = `"<div class=\\"lo--select-avatar--table--body--item access-token \\"><input readonly=\\"\\" type=\\"text\\" value=\\"token\\"><div class=\\"renew\\">Renew</div></div>"`;
